examples/src/main/java/com/cloudera/nav/sdk/examples/lineage3/LineageExport.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2016 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cloudera.nav.sdk.examples.lineage3;
+
+import com.cloudera.nav.sdk.client.ClientConfig;
+import com.cloudera.nav.sdk.client.MetadataQuery;
+import com.cloudera.nav.sdk.client.NavApiCient;
+import com.cloudera.nav.sdk.client.NavigatorPlugin;
+import com.cloudera.nav.sdk.client.ResultsBatch;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.net.util.Base64;
+
+/**
+ * This example shows how the lineage3 export API can be used to fetch
+ * lineage for a entities retrieved as a result of issuing a query. The
+ * lineage is streamed from the Navigator server and is written into a
+ * lineage.json file, that is created in the current working directory.
+ * The example calculates lineage downstream with a length of 5. It uses
+ * zero as the value of lineageOptions which applies all the filters.
+ * <p/>
+ * Note that the following navigator options have to be set before this
+ * example program can work.
+ * <p/>
+ * 1. navigator.api.lineage.export_enabled
+ * Set it to true to enable the export API
+ * <p/>
+ * 2. navigator.api.lineage.export_batch_size
+ * This sets the maximum number of entities for which the
+ * lineage could be requested for. By default it is set to 100.
+ */
+
+public class LineageExport {
+
+  private String configPath;
+  private String query;
+  private String direction;
+  private String length;
+  private String lineageOptions;
+  private final static String LINEAGE_EXPORT_API = "lineage3/export";
+  private final static String LINEAGE_EXPORT_API_QUERY_PARAMS = "?direction=%s&length=%s&lineageOptions=%s";
```

Comment:
Yup I saw that, the reason I didn't use that is to avoid creating a class which mimics the structure of the array of the elements that are sent back as the response from the API. Using mapper and Object.class creates ArrayLists and LinkedHashMaps and I use them to extract out the various things I need to print out in the file.
I thought it would be upto the user however they want to massage the results they receive from the API. If you feel strongly about it, I can create a class and then just use the RestTemplate.

Yes it is now a POST request.